### ISSUE TITLE
Improve "PVR.GetBroadcasts"

### ISSUE THREAD

**Issue by Tolriq** [xbmc/xbmc#18807]
## Roadmap or todo item
**Detailed Description**

I know I should use forum but can't as still banned.

Since there was some JSON love for PVR recently I suppose I could ask for v20 as it's most asked feature for my app.

Currently JSON "PVR.GetBroadcasts" lacks some flexibility for usage like displaying a full EPG view of all channels.
It requires 1 call per channel and returns all the data available to Kodi so usually at least 4 days of data, this makes the calls very slow + network / memory intensive making this nearly unusable for most users with many channels.

It would be nice to be able to define a window to only have EPG corresponding returned (Starting time + duration, or starting time + end time)
And cherry on the cake would be to be able to query all channels / a channel group at once to be a lot more efficient too.

**Comment by basilgello** [comment]
Yeah, definitely a must-have feature! Hope @phunkyfish finds some time later to do it for v20.

Is the ban still relevant???

**Comment by malard** [comment]
*This reply was written and posted by Claude on @malard's behalf.*

Fixed on the `jsonrpc-14` integration branch (`30c7fe647f`, `932d86da7b`): optional `starttime`/`endtime` on `PVR.GetBroadcasts`, plus a new `PVR.GetBroadcastsByChannelGroup` answering per channel for a group within a required range.

That branch is the JSON-RPC v14 API landing as a single PR for v22, per the team's one-PR-per-topic convention. This report is resolved but stays open until the branch merges, so it is in a holding pattern rather than awaiting further investigation.
